### PR TITLE
Gurzeh 4553

### LINKED ISSUE #4553: Wrong level_stats calculation
### Expected Behavior
at any level, level_stats should show (like the app):
XP earned at the current level / XP to pass next level

### Actual Behavior
in fact, level_apps shows (XP previous level + XP earned at the current level) / (XP previous level + XP to pass next level)
Ex: Level 16 is 20.000 and level 17 also, so when it starts, it shows 50% where it should show 0%
Level 16 (20,095 / 40,000, 50%)

### Your FULL config.json (remove your username, password, gmapkey and any other private info)
http://pastebin.com/XzFy46Tv

### Output when issue occurred
2016-08-22 14:57:34,214 [UpdateLiveStats] [INFO] [log_stats] XXX | Uptime : 0:19:01 | Earned 2,200 Stardust | +4,500 XP | 14,200 XP/h | Visited 15 stops | Level 16 (20,095 / 40,000, **50%**)

### Other Information
OS: OSX Capitan
Branch: dev
Git Commit: 4f2e976764242a69e4ed7551ceae805d19f3a55c
Python Version: Python 2.7.12

### PR DESCRIPTION
**Pull request by Gurzeh**
## Short Description:
As current_level_xp is not available in API, we put hard coded values in matrix xp_per_level and referred to it to calculate accurate stats.

## Details
at any level, level_stats should show (like the app): 
XP earned at the current level / XP to pass next level

in fact, level_apps shows (XP previous level + XP earned at the current level) / (XP previous level + XP to pass next level)
Ex: Level 16 is 20.000 and level 17 also, so when it starts, it shows 50% where it should show 0%
Level 16 (20,095 / 40,000, 50%)

## Issues references
Fixes #4553

### CODE REVIEW

File: pokemongo_bot/cell_workers/update_live_stats.py [hunk 1]
```diff
@@ -172,8 +172,14 @@ def _update_title(self, title, platform):
             self.terminal_title = False
 
         self._compute_next_update()
-
+    
+    # hard coded values to supplement unknown current_level_xp
+    global xp_per_level
+    xp_per_level = [[0, 0], [1000, 1000], [2000, 3000], [3000, 6000], [4000, 10000], [5000, 15000], [6000, 21000], [7000, 28000], [8000, 36000], [9000, 45000], [10000, 55000], [10000, 65000], [10000, 75000], [10000, 85000], [15000, 100000], [20000, 120000], [20000, 140000], [20000, 160000], [25000, 185000], [25000, 210000], [50000, 260000], [75000, 335000], [100000, 435000], [125000, 560000], [150000, 710000], [190000, 900000], [200000, 1100000], [250000, 1350000], [300000, 1650000], [350000, 2000000], [500000, 2500000], [500000, 3000000], [750000, 3750000], [1000000, 4750000], [1250000, 6000000], [1500000, 7500000], [2000000, 9500000], [2500000, 12000000], [3000000, 15000000], [5000000, 20000000]]
```
Comment (th3w4y) [inline pokemongo_bot/cell_workers/update_live_stats.py]: how do i read this? is there any pattern?

Comment (Gurzeh) [inline pokemongo_bot/cell_workers/update_live_stats.py]: These are couples of xp / level (40 of them)
The first value is the XP to reach the level, the second is the total XP earned.
Reference is there http://heavy.com/games/2016/07/pokemon-go-xp-gained-per-level-how-much-needed-need-to-up-advance-chart-graph-image/

**Comment by Gurzeh** [comment]
It seems the tests/update_live_stats_test must be updated as well.

**Comment by k4n30** [comment]
@Gurzeh Are you going to take care of updating the test as well?

Also the xp_per_level data should probably live in the [data folder](https://github.com/PokemonGoF/PokemonGo-Bot/tree/dev/data) rather than the function so that if any other workers require access to it, they can.  I agree it may not update very often, but it is global data that should be able to be used by others (good coding practice)

**Comment by Gurzeh** [comment]
@k4n30 yes for updating the test, and ok for moving the list outside code.

**Comment by Gurzeh** [comment]
@k4n30 not that familiar with GitHub... But I think the new commits are incorporated. Do you confirm?

**Comment by solderzzc** [comment]
@k4n30 @Gurzeh  This is good for now. Time to merge.

**Comment by k4n30** [comment]
Looks good to me :) added second review tag for devs to review and merge if all good

**Pull request merged**